python/AIEMLIRModule.cpp

```diff
@@ -101,17 +102,20 @@ PYBIND11_MODULE(_aie, m) {
         if (mlirLogicalResultIsFailure(aieTranslateToCDODirect(
                 op, {workDirPath.data(), workDirPath.size()}, bigendian,
                 emitUnified, cdoDebug, aieSim, xaieDebug, enableCores)))
-          throw py::value_error("Failed to generate cdo because: " +
-                                scope.takeMessage());
+          throw nb::value_error(
+              (llvm::Twine("Failed to generate cdo because: ") +
+               llvm::Twine(scope.takeMessage()))
+                  .str()
+                  .c_str());
       },
       "module"_a, "work_dir_path"_a, "bigendian"_a = false,
       "emit_unified"_a = false, "cdo_debug"_a = false, "aiesim"_a = false,
       "xaie_debug"_a = false, "enable_cores"_a = true);
 
   m.def(
       "transaction_binary_to_mlir",
-      [](MlirContext ctx, py::bytes bytes) {
-        std::string s = bytes;
+      [](MlirContext ctx, nb::bytes bytes) {
+        std::string s = static_cast<const char *>(bytes.data());
         MlirStringRef bin = {s.data(), s.size()};
```

Comment:
```suggestion
        MlirStringRef bin = {static_cast<const char *>(bytes.data()),
                             bytes.size()};
```
For me this fixed the txn2mlir testing error